internal/handlers/common/filter.go

```diff
@@ -58,31 +58,52 @@ func filterDocumentPair(doc *types.Document, filterKey string, filterValue any)
 		// {field1./.../.fieldN: filterValue}
 		path := types.NewPathFromString(filterKey)
 		// we pass the path without the last key because we want {fieldN: *someValue*}, not just *someValue*
-		docValue, err := doc.GetByPath(path.TrimSuffix())
+		docValues, err := doc.GetAllByPath(path.TrimSuffix())
 		if err != nil {
 			return false, nil // no error - the field is just not present
 		}
 
-		switch docValue := docValue.(type) {
-		case *types.Document:
-			doc = docValue
-			filterKey = path.Suffix()
-		case *types.Array:
-			index, err := strconv.Atoi(path.Suffix())
-			if err != nil {
-				return false, nil
-			}
+		for _, docValue := range docValues {
+			switch docValue := docValue.(type) {
+			case *types.Document:
+				doc = docValue
+				filterKey = path.Suffix()
+			case *types.Array:
+				p := path.Suffix()
+				index, err := strconv.Atoi(p)
+
+				switch err {
```

Comment:
Hmm, switch with `case nil:` and `default:` seems like it can be `if err != nil` :innocent: